src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,123 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
+type Props = {
+  onAdd: (currentState: any) => void,
+};
+
+export const NewMovie:React.FC<Props> = ({
+  onAdd,
+}) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+  const [objMovie, setObjMovie] = useState({
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  });
+
+  const areAllFieldsFilledIn = (obj: Movie) => {
+    if (
+      obj.title === ''
+      || obj.imdbId === ''
+      || obj.imdbUrl === ''
+      || obj.imgUrl === '') {
+      return true;
+    }
+
+    return false;
+  };
 
   return (
-    <form className="NewMovie" key={count}>
-      <h2 className="title">Add a movie</h2>
+    <form
+      className="NewMovie"
+      key={count}
+      onSubmit={(event) => {
+        event.preventDefault();
+        if (areAllFieldsFilledIn(objMovie)) {
+          return;
+        }
+
+        setCount(count + 1);
 
+        setObjMovie({
+          title: '',
+          description: '',
+          imgUrl: '',
+          imdbUrl: '',
+          imdbId: '',
+        });
+
+        onAdd(objMovie);
+      }}
```

Comment:
create separate function `onSubmit` and move the code there
```suggestion
      onSubmit={onSubmit}
```